src/burn/engine/plan.cpp

```diff
@@ -1063,16 +1067,21 @@ extern "C" HRESULT PlanCachePackage(
 
     if (fBARequestedCache || NeedsCache(pPlan, pPackage))
     {
+        // The behavior for cache only mode is to do nothing on rollback, e.g. for subsequent install on demand scenarios.
+        pPackage->rollback = BOOTSTRAPPER_ACTION_STATE_NONE;
+
         hr = AddCachePackage(pPlan, pPackage, phSyncpointEvent);
         ExitOnFailure(hr, "Failed to plan cache package.");
 
+        // Update plan state to account for the package being cached
+        pPlan->qwEstimatedSize += pPackage->qwSize;
```

Comment:
I'm guessing this needs to be changed in wix3 as well? Seems qwEstimatedSize is updated elsewhere there as well.